### PR TITLE
REM-395

### PR DESCRIPTION
**Pull request by imtithalhalalo**
page name = “sa-financial-integrator”
This is an inline grid, where 1st column is readOnly, 2nd column is editable/optional combobox & third column is checkbox.
On submit should not pass the whole inline rows, set each row separately.

### CODE REVIEW

File: src/pages/sa-financial-integrator/index.js [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+            postRequest({
+              extension: SaleRepository.FinancialIntegrators.set,
+              record: JSON.stringify(row)
+            })
+          )
+        )
+
+        toast.success(platformLabels.Updated)
+        await getGridData()
+      } catch (error) {}
```
Comment (RebeccaFrem02) [inline src/pages/sa-financial-integrator/index.js:63]: can we try removing empty try-catch and see if anything will break?

File: src/pages/sa-financial-integrator/index.js [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    queryFn: getGridData,
+    datasetId: ResourceIds.FinancialIntegrators
+  })
+
+  const { formik } = useForm({
+    maxAccess,
+    enableReinitialize: true,
+    validateOnChange: true,
+    initialValues: {
+      rows: []
```
Comment (RebeccaFrem02) [inline src/pages/sa-financial-integrator/index.js:48]: shouldn't you add cgId, fgId and sameNumber in the rows in the initial values as null?

File: src/pages/sa-financial-integrator/index.js [hunk 3]
```diff
@@ -0,0 +1,127 @@
+import { useContext } from 'react'
+import toast from 'react-hot-toast'
+import { RequestsContext } from 'src/providers/RequestsContext'
+import { SystemRepository } from 'src/repositories/SystemRepository'
```
Comment (omarZrayka23) [inline src/pages/sa-financial-integrator/index.js:4]: clear , it is not used

File: src/pages/sa-financial-integrator/index.js [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+import { useForm } from 'src/hooks/form'
+import FormShell from 'src/components/Shared/FormShell'
+import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
+import { Grow } from 'src/components/Shared/Layouts/Grow'
+import { DataGrid } from 'src/components/Shared/DataGrid'
+import { GeneralLedgerRepository } from 'src/repositories/GeneralLedgerRepository'
+import { ControlContext } from 'src/providers/ControlContext'
+import { Fixed } from 'src/components/Shared/Layouts/Fixed'
+import WindowToolbar from 'src/components/Shared/WindowToolbar'
+import { Formik } from 'formik'
```
Comment (omarZrayka23) [inline src/pages/sa-financial-integrator/index.js:16]: clear , it is not used

Comment (omarZrayka23) [inline src/pages/sa-financial-integrator/index.js:16]: clear , it is not used

File: src/pages/sa-financial-integrator/index.js [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+import { RequestsContext } from 'src/providers/RequestsContext'
+import { SystemRepository } from 'src/repositories/SystemRepository'
+import { useResourceQuery } from 'src/hooks/resource'
+import { ResourceIds } from 'src/resources/ResourceIds'
+import { useForm } from 'src/hooks/form'
+import FormShell from 'src/components/Shared/FormShell'
+import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
+import { Grow } from 'src/components/Shared/Layouts/Grow'
+import { DataGrid } from 'src/components/Shared/DataGrid'
+import { GeneralLedgerRepository } from 'src/repositories/GeneralLedgerRepository'
```
Comment (omarZrayka23) [inline src/pages/sa-financial-integrator/index.js:12]: clear , it is not used

**Review by omarZrayka23 — APPROVED** [review]

**Pull request merged**